### PR TITLE
Change base directory of socket path via SERVERENGINE_SOCKETMANAGER_SOCK_DIR env

### LINKED ISSUE #86: linux: allow configurable socketmanager path
Currently hardwired:

https://github.com/treasure-data/serverengine/blob/master/lib/serverengine/socket_manager.rb#L71

 '/tmp/SERVERENGINE_SOCKETMANAGER_' + Time.now.to_s.gsub(' ', '') + '_' + Process.pid.to_s

which can cause problems if /tmp is periodically cleaned up

I encountered this using fluentd, so arguably it's a issue there, rather than serverengine:

https://github.com/fluent/fluentd/blob/master/lib/fluent/supervisor.rb#L57

### PR DESCRIPTION
**Pull request by repeatedly**
fix #86

### CODE REVIEW

**Comment by repeatedly** [comment]
@tagomoris Could you check this?

File: lib/serverengine/socket_manager.rb [hunk 1]
```diff
@@ -68,7 +68,8 @@ def self.generate_path
             end
           end
         else
-          '/tmp/SERVERENGINE_SOCKETMANAGER_' + Time.now.to_s.gsub(' ', '') + '_' + Process.pid.to_s
+          base_dir = (ENV['SERVERENGINE_SOCKETMANAGER_SOCK_DIR'] || '/tmp')
+          File.join(base_dir, 'SERVERENGINE_SOCKETMANAGER_' + Time.now.to_s.gsub(' ', '') + '_' + Process.pid.to_s)
```
Comment (tagomoris) [inline lib/serverengine/socket_manager.rb]: How about to switch to use `Time.now.utc.iso8601`?
It creates timezone-independent path and doesn't contain whitespaces, like: `2017-03-01T16:14:12+09:00`.

Comment (repeatedly) [inline lib/serverengine/socket_manager.rb]: It looks good. Will change.

**Comment by repeatedly** [comment]
Apply review and fix rake version for tests.

TODO: Use rspec3 for rake v12.

**Comment by tagomoris** [comment]
LGTM.

**Pull request merged**